packages/expo/ios/Fetch/ExpoNetworkFetchModule.swift

```diff
@@ -16,12 +17,8 @@ public final class ExpoNetworkFetchModule: Module {
   public func definition() -> ModuleDefinition {
     Name("ExpoNetworkFetchModule")
 
-    OnCreate {
-      urlSession = createDefaultURLSession()
-    }
-
     OnDestroy {
-      urlSession?.invalidateAndCancel()
+      urlSession.invalidateAndCancel()
```

Comment:
What if the module doesn't create the session? I liked the version without lazy more. That can lead to unnecessary creation of session during modules destruction.

Reply:
since `EXVersionedNetworkInterceptor` is set up after module OnCreate, we have to delay the URLSession creation. if the URLSession doesn't create (i.e. no request sent at all), here OnDestroy would just create URLSession and invalidate. it's not 100% ideal but that's so far the cleanest code i can tell.